mindsdb/interfaces/storage/db.py

```diff
@@ -0,0 +1,79 @@
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column, Integer, String, Float, DateTime, Boolean, MetaData
+from sqlalchemy.schema import ForeignKey
+import datetime
+
+if os.environ['MINDSDB_DATABASE_TYPE'] == 'sqlite':
+    engine = create_engine('sqlite:///' + os.environ['MINDSDB_SQLITE_PATH'], echo=False) # + '?check_same_thread=False'
+elif os.environ['MINDSDB_DATABASE_TYPE'] == 'mariadb':
+    raise Exception('Mariadb not supported !')
+
+Base = declarative_base()
+session = scoped_session(sessionmaker(bind=engine,autoflush=True))
+Base.query = session.query_property()
+entitiy_version = 1
+
+class Semaphor(Base):
+    __tablename__ = 'semaphor'
+
+    id = Column(Integer, primary_key=True)
+    modified_at = Column(DateTime, default=datetime.datetime.now, onupdate=datetime.datetime.now)
+    created_at = Column(DateTime, default=datetime.datetime.now)
+    entity_type = Column(String)
+    entity_id = Column(String)
+
+class Configuration(Base):
+    __tablename__ = 'configuration'
+
+    id = Column(Integer, primary_key=True)
+    modified_at = Column(DateTime, default=datetime.datetime.now, onupdate=datetime.datetime.now)
+    created_at = Column(DateTime, default=datetime.datetime.now)
+    data = Column(String) # A JSON
+    company_id = Column(Integer, unique=True)
+
+class Datasource(Base):
+    __tablename__ = 'datasource'
+
+    id = Column(Integer, primary_key=True)
+    modified_at = Column(DateTime, default=datetime.datetime.now, onupdate=datetime.datetime.now)
+    created_at = Column(DateTime, default=datetime.datetime.now)
+    name = Column(String)
+    data = Column(String) # Including, e.g. the query used to create it and even the connection info when there's no integration associated with it -- A JSON
+    analysis = Column(String)  # A JSON
+    company_id = Column(Integer)
+    version = Column(Integer, default=entitiy_version)
+    integration_id = Column(Integer)
+
+
+class Predictor(Base):
+    __tablename__ = 'predictor'
+
+    id = Column(Integer, primary_key=True)
+    modified_at = Column(DateTime, default=datetime.datetime.now, onupdate=datetime.datetime.now)
+    created_at = Column(DateTime, default=datetime.datetime.now)
+    name = Column(String)
+    data = Column(String) # A JSON
+    native_version = Column(String)
+    to_predict = Column(String)
+    status = Column(String)
+    company_id = Column(Integer)
+    version = Column(Integer, default=entitiy_version)
+    datasource_id = Column(Integer, ForeignKey('datasource.id'))
+    is_custom = Column(Boolean)
+
+class Log(Base):
+    __tablename__ = 'log'
+
+    id = Column(Integer, primary_key=True)
+    created_at = Column(DateTime, default=datetime.datetime.now)
+    log_type = Column(String) # log, info, warning, traceback etc
+    source = Column(String) # file + line
+    company_id = Column(Integer)
+    payload = Column(String)
+
+
+Base.metadata.create_all(engine)
```

Comment:
If config['storage_dir'] is not exists on mindsdb start, then i get error here:
```
Traceback (most recent call last):
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 2336, in _wrap_pool_connect
    return fn()
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 364, in connect
    return _ConnectionFairy._checkout(self)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 778, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 495, in checkout
    rec = pool._do_get()
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/impl.py", line 241, in _do_get
    return self._create_connection()
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 309, in _create_connection
    return _ConnectionRecord(self)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 440, in __init__
    self.__connect(first_connect_check=True)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 661, in __connect
    pool.logger.debug("Error on connect(): %s", e)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    with_traceback=exc_tb,
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 656, in __connect
    connection = pool._invoke_creator(self)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/strategies.py", line 114, in connect
    return dialect.connect(*cargs, **cparams)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 493, in connect
    return self.dbapi.connect(*cargs, **cparams)
sqlite3.OperationalError: unable to open database file

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/maxs/.vscode/extensions/ms-python.python-2021.1.502429796/pythonFiles/lib/python/debugpy/__main__.py", line 45, in <module>
    cli.main()
  File "/home/maxs/.vscode/extensions/ms-python.python-2021.1.502429796/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 444, in main
    run()
  File "/home/maxs/.vscode/extensions/ms-python.python-2021.1.502429796/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 331, in run_module
    run_module_as_main(target_as_str, alter_argv=True)
  File "/usr/lib/python3.6/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib/python3.6/runpy.py", line 142, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/usr/lib/python3.6/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/home/maxs/dev/mdb/venv_new/sources/mindsdb/mindsdb/__init__.py", line 72, in <module>
    from mindsdb.utilities.config import Config
  File "/home/maxs/dev/mdb/venv_new/sources/mindsdb/mindsdb/utilities/config.py", line 8, in <module>
    from mindsdb.interfaces.storage.db import session, Configuration
  File "/home/maxs/dev/mdb/venv_new/sources/mindsdb/mindsdb/interfaces/storage/db.py", line 78, in <module>
    Base.metadata.create_all(engine)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", line 4554, in create_all
    ddl.SchemaGenerator, self, checkfirst=checkfirst, tables=tables
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 2094, in _run_visitor
    with self._optional_conn_ctx_manager(connection) as conn:
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 2086, in _optional_conn_ctx_manager
    with self._contextual_connect() as conn:
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 2302, in _contextual_connect
    self._wrap_pool_connect(self.pool.connect, None),
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 2340, in _wrap_pool_connect
    e, dialect, self
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1584, in _handle_dbapi_exception_noconnection
    sqlalchemy_exception, with_traceback=exc_info[2], from_=e
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 2336, in _wrap_pool_connect
    return fn()
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 364, in connect
    return _ConnectionFairy._checkout(self)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 778, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 495, in checkout
    rec = pool._do_get()
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/impl.py", line 241, in _do_get
    return self._create_connection()
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 309, in _create_connection
    return _ConnectionRecord(self)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 440, in __init__
    self.__connect(first_connect_check=True)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 661, in __connect
    pool.logger.debug("Error on connect(): %s", e)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    with_traceback=exc_tb,
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 656, in __connect
    connection = pool._invoke_creator(self)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/strategies.py", line 114, in connect
    return dialect.connect(*cargs, **cparams)
  File "/home/maxs/dev/mdb/venv_new/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 493, in connect
    return self.dbapi.connect(*cargs, **cparams)
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file
(Background on this error at: http://sqlalche.me/e/13/e3q8)
```

Reply:
Strange, does this happen you have a pre-existing config file but this file *doesn't* have a storage path?

Reply:
I don't get this error. Whether I use no config or a config without a `sotrage_path` can you provide the config used?